CONTRIBUTING.md

```diff
@@ -81,19 +142,119 @@ If you're working inside of a CVAT environment, you can also use:
+we use special annotations for tests.
 
-## Environment
+A test needs to marked with a requirement it is related to. To mark a test, use:
+
+```python
+from unittest import TestCase
+from .requirements import Requirements, mark_requirement
+
+class MyTests(TestCase):
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_my_requirement(self):
+        ... do stuff ...
+```
+
+Such marking will apply markings from the requirement specified.
+They can be overriden for a specific test:
+
+```python
+import pytest
+
+    @pytest.mark.proirity_low
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_my_requirement(self):
+        ... do stuff ...
+```
+
+#### Requirements <a id="Requirements"></a>
+
+Requirements need to be added to [`tests/requirements.py`](tests/requirements.py):
+
+```python
+DATUM_BUG_219 = "Return format is not uniform"
+DATUM_244 = "Add Snyk integration"
+```
+
+```python
+# Fully defined in GitHub issues:
+@pytest.mark.reqids(Requirements.DATUM_244, Requirements.DATUM_333)
+
+# And defined ony other way:
+@pytest.mark.reqids(Requirements.DATUM_GENERAL_REQ)
+```
+
+##### Available annotations for tests and requirements
+
+Markings are defined in [`tests/conftest.py`](tests/conftest.py).
+
+**A list of requirements**
+```python
+@pytest.mark.requids(Requirements.DATUM_123)
+```
+
+**A priority**
+```python
+@pytest.mark.priority_low
+@pytest.mark.priority_medium
+@pytest.mark.priority_high
+```
+
+**Component**
+The marking used for indication of different system components
+
+```python
+@pytest.mark.components(DatumaroComponent.Datumaro)
+```
+
+**Skipping tests**
+
+```python
+@pytest.mark.skip(SkipMessages.NOT_IMPLEMENTED)
+```
+
+**Parametrized runs**
+
+Parameters are used for running the same test with different parameters e.g.
+
+```python
+@pytest.mark.parametrize("numpy_array, batch_size", [
+    (np.zeros([2]), 0),
+    (np.zeros([2]), 1),
+    (np.zeros([2]), 2),
+    (np.zeros([2]), 5),
+    (np.zeros([5]), 2),
+])
+```
+
+### Test documentation <a id="TestDoc"></a>
+
+Tests are documented with docstrings. Test descriptions must contain
+the following: sections: `Description`, `Expected results` and `Steps`.
+
+```python
+def test_can_convert_polygons_to_mask(self):
+    """
+    <b>Description:</b>
+    Ensure that the dataset polygon annotation can be properly converted
+    into dataset segmentation mask.
+
+    <b>Expected results:</b>
+    Dataset segmentation mask converted from dataset polygon annotation
+    is equal to an expected mask.
 
-The recommended editor is VS Code with the Python plugin.
\ No newline at end of file
+    <b>Steps:</b>
+    1. Prepare dataset with polygon annotation
+    2. Prepare dataset with expected mask segmentation mode
+    3. Convert source dataset to target, with conversion of annotation
+      from polygon to mask.
+    4. Verify that resulting segmentation mask is equal to the expected mask.
+    """
```

Comment:
@sstrehlk, please check if step comments are really needed.

Reply:
We can resign of steps comments in test body .